docs/advanced/input_files/input-main.md

```diff
@@ -3992,4 +3995,20 @@ The output files are `OUT.${suffix}/Excitation_Energy.dat` and `OUT.${suffix}/Ex
 - **Description**: Atomic basis set size for each atom type (with the same order as in `STRU`) in FHI-aims.
 - **Default**: {} (empty list, where ABACUS use its own basis set size)
 
+## Reduced Density Matrix Functional Theory
+
+ab-initio methods and the xc-functional parameters used in RDMFT.
+
+### ab_initio_type
```

Comment:
Hi, there is another keyword `esolver_type`, and `ksdft` is a possible value of it, could you check if `rdmft` can be a param of it?

Reply:
theoretically, rdmft can be a value of esolver_type.

But there are too many places in the program that involve esolver_type, which makes me think that it is dangerous to use esolver_type to express rdmft   (when calculating rdmft, we usually need to run one or several steps of ksdft calculation to get the initial value, so when running rdmft, we must ensure that the ksdft program can run completely normally)

So I temporarily added the parameter ab_initio_type

Reply:
Thank you for your opinion, senior sister.

In fact, the theoretical status of RDMFT and DFT is the same, but when we apply RDMFT in practice, we can use the results of one or several steps of KS as the initial value (of course, we can also use a random 1-RDM as the initial value), which requires KSDFT to run completely in the actual program.

However, in the current RDMFT implementation, I am still writing the program for variational optimization to the 1-RDM of the ground state (the optimization program of the cooperating teacher is not fully written yet), so it seems that the code I pushed up is doing some "repetitive" calculations without practical significance except for using a few new functionals.

Professor Ren meant that, because ABACUS is often refactored, I need to spend a long time to rewrite the correct rdmft code in the old version into the new version every once in a while, which is a waste of time and has no practical significance for the development of new methods. Therefore, it is recommended that I merge the rdmft code into the developer branch in time, but not release it in the user-oriented version for the time being (as long as my code itself does not affect the normal code of ABACUS, for example, when the ab_initio_type of INPUT is the default "ksdft", my code has no effect on other calculations of ABACUS), so the current rdmft module is still considered a production version.

The physical quantities that rdmft temporarily expects to output are the kinetic energy, total energy and 1-RDM of the system in the ground state (by the way, where is this document)

Reply:
Thank you for your opinion, senior sister.

In fact, the theoretical status of RDMFT and DFT is the same, but when we apply RDMFT in practice, we can use the results of one or several steps of KS as the initial value (of course, we can also use a random 1-RDM as the initial value), which requires KSDFT to run completely in the actual program.

However, in the current RDMFT implementation, I am still writing the program for variational optimization to the 1-RDM of the ground state (the optimization program of the cooperating teacher is not fully written yet), so it seems that the code I pushed up is doing some "repetitive" calculations without practical significance except for using a few new functionals.

Professor Ren meant that, because ABACUS is often refactored, I need to spend a long time to rewrite the correct rdmft code in the old version into the new version every once in a while, which is a waste of time and has no practical significance for the development of new methods. Therefore, it is recommended that I merge the rdmft code into the developer branch in time, but not release it in the user-oriented version(as long as my code itself does not affect the normal code of ABACUS, for example, when the ab_initio_type of INPUT is the default "ksdft", my code has no effect on other calculations of ABACUS), so the current rdmft module is still considered a production version.

The physical quantities that rdmft temporarily expects to output are the kinetic energy, total energy and 1-RDM of the system in the ground state (by the way, where is this document)

Reply:
I see. I mean such information can appear in the parameter description in this .md file.
> The physical quantities that rdmft temporarily expects to output are the kinetic energy, total energy and 1-RDM of the system in the ground state (by the way, where is this document)

Reply:
I would suggest changing " ab_initio_type" to "rdmft" and set its type to bool, and make this variable controlled by yourself. Do not invent any ambiguous variable for other developers to think they can do something new with this variable.